src/libponyrt/mem/pool.c

```diff
@@ -971,7 +971,7 @@ void* ponyint_pool_realloc_size(size_t old_size, size_t new_size, void* p)
     new_p = pool_alloc_size(new_adj_size);
   }
 
-  memcpy(new_p, p, new_size);
+  memcpy(new_p, p, old_size);
```

Comment:
Good catch!

But is it guaranteed that `old_size < new_size` here?

If not, wouldn't we want to take `min(old_size, new_size)` as the copy length?